### PR TITLE
ESSNTL-4324: Enable IQE tests in the PR check

### PR DESCRIPTION
**Pull request by fstavela**
[ESSNTL-4324](https://issues.redhat.com/browse/ESSNTL-4324): Run IQE tests in the PR check.

### CODE REVIEW

**Comment by fstavela** [comment]
@fijshion @skateman Please review

**Comment by skateman** [comment]
Hmm, it does not save the artifacts from the ephemeral deployment + test. It either doesn't run or it just forgets about uploading.

**Comment by fstavela** [comment]
> Hmm, it does not save the artifacts from the ephemeral deployment + test. It either doesn't run or it just forgets about uploading.

@skateman  Interesting, thanks for catching that. I finally figured out what's happening and I think we have 2 issues here.

But first of all, the ephemeral deployment is successful and the tests are successfully running. The artifacts are collected and they are saved. If you open the test job, you can find them there.

Now, the first issue is that we are unable to upload the artifacts to the S3 bucket and so the test results are unavailable in the Ibutsu. This is happening for all of the services, including host-inventory, compliance etc. So if you look at any PR on any of the Insights services and open the "Test results are available in Ibutsu" message in the PR checks, it will open Ibutsu and say "Run not found". I think this is caused by this error I found in the pipeline logs:
```
13:57:17 ���� ���INFO: uploading artifacts to S3
13:57:17 upload-artifact-s3: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by upload-artifact-s3)
13:57:17 upload-artifact-s3: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by upload-artifact-s3)
13:57:17 ��� ERROR: Failed to upload artifacts to S3
```
I think the root cause is similar to what has been causing the host-inventory pre-commit to fail in the PR check - app-sre changed the jenkins worker image and so we probably miss some libraries. I will open a ticket for this, but I don't think this is something we can/should fix on our end.

The second issue is that we don't even see that ibutsu message in our PRs in this repo. This is actually the same for xjoin-search. I've been digging into this and I found out that when we try to POST to the GitHub API and create that message, we get this response:
```
{"message":"Not Found","documentation_url":"https://docs.github.com/rest/commits/statuses#create-a-commit-status"}
```
What I think is happening is that the `InsightsDroid` doesn't have push access for the commit messages in these repos. Unfortunately, I'm not sure, because the GitHub error message is very vague and if I'm right then it's actually not correct. Since this is a public repo, it should say something like "Permissions denied" instead of "Not Found". I don't have permissions to check or fix this. @skateman can you please take a look in the repo settings and give permissions to InsightsDroid so we can check if that helps?

**Comment by fijshion** [comment]
@fstavela I gave the QE group (which contains InsightsDroid) write access to this repo. I'll go through all the other xjoin repos and do the same.

File: clowdapp.yaml [hunk 1]
```diff
@@ -9,8 +9,13 @@ objects:
       name: xjoin-api-gateway
     spec:
       envName: ${ENV_NAME}
+      testing:
+        iqePlugin: xjoin
       dependencies:
         - xjoin-apicurio
+      optionalDependencies:
+        - xjoin-search
+        - host-inventory
```
Comment (fijshion) [inline clowdapp.yaml:18]: Also wondering with host-inventory is required? The purpose of xjoin.v2 is to make xjoin generic so it shouldn't depend on host-inventory.

**Comment by Victoremepunto** [comment]
> > > Hmm, it does not save the artifacts from the ephemeral deployment + test. It either doesn't run or it just forgets about uploading.
> > 
> > 
> > @skateman Interesting, thanks for catching that. I finally figured out what's happening and I think we have 2 issues here.
> > But first of all, the ephemeral deployment is successful and the tests are successfully running. The artifacts are collected and they are saved. If you open the test job, you can find them there.
> > Now, the first issue is that we are unable to upload the artifacts to the S3 bucket and so the test results are unavailable in the Ibutsu. This is happening for all of the services, including host-inventory, compliance etc. So if you look at any PR on any of the Insights services and open the "Test results are available in Ibutsu" message in the PR checks, it will open Ibutsu and say "Run not found". I think this is caused by this error I found in the pipeline logs:
> > ```
> > 13:57:17 ���� ���INFO: uploading artifacts to S3
> > 13:57:17 upload-artifact-s3: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by upload-artifact-s3)
> > 13:57:17 upload-artifact-s3: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by upload-artifact-s3)
> > 13:57:17 ��� ERROR: Failed to upload artifacts to S3
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I think the root cause is similar to what has been causing the host-inventory pre-commit to fail in the PR check - app-sre changed the jenkins worker image and so we probably miss some libraries. I will open a ticket for this, but I don't think this is something we can/should fix on our end.
> > _EDIT: The first issue is caused by IQE moving to ubi-minimal image. More info here: https://issues.redhat.com/browse/IQE-2057_
> > The second issue is that we don't even see that ibutsu message in our PRs in this repo. This is actually the same for xjoin-search. I've been digging into this and I found out that when we try to POST to the GitHub API and create that message, we get this response:
> > ```
> > {"message":"Not Found","documentation_url":"https://docs.github.com/rest/commits/statuses#create-a-commit-status"}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > What I think is happening is that the `InsightsDroid` doesn't have push access for the commit messages in these repos. Unfortunately, I'm not sure, because the GitHub error message is very vague and if I'm right then it's actually not correct. Since this is a public repo, it should say something like "Permissions denied" instead of "Not Found". I don't have permissions to check or fix this. @skateman can you please take a look in the repo settings and give permissions to InsightsDroid so we can check if that helps?
> 
> @Victoremepunto maybe you can give us a hand here? thinking

The @InsightsDroid , afaik, should have permissions as is part of the Insights organization - maybe I'm wrong, I'll have to look into this, but should only mean that the droid is not able to report back to the PR and add a message with the link (maybe probably because of the 1st issue?) .

File: clowdapp.yaml (hunk 1 above)
Comment (fstavela) [reply clowdapp.yaml:18]: Same answer as above. It's only optional, which means it's only needed for testing.

**Comment by fstavela** [comment]
@fijshion @skateman From my side, this is ready to be merged.

**Review by fijshion — APPROVED** [review]

**Pull request merged**